main.go

```diff
@@ -13,13 +13,29 @@ func setupServer() *server.MCPServer {
 		"Asgardeo Management MCP",
 		"0.0.1",
 		server.WithResourceCapabilities(true, true),
+		server.WithToolCapabilities(true),
 		server.WithLogging(),
 		server.WithRecovery(),
 	)
 
 	appListTool, appListToolImpl := tools.GetApplicationListTool()
 	s.AddTool(appListTool, appListToolImpl)
 
+	spaTool, spaToolImpl := tools.GetSinglePageAppTool()
+	s.AddTool(spaTool, spaToolImpl)
+
+	mobileAppTool, mobileAppToolImpl := tools.GetMobileAppTool()
+	s.AddTool(mobileAppTool, mobileAppToolImpl)
+
+	m2mAppTool, m2mAppToolImpl := tools.GetM2MAppTool()
+	s.AddTool(m2mAppTool, m2mAppToolImpl)
+
+	getAppByNameTool, getAppByNameToolmpl := tools.GetApplicationByNameTool()
+	s.AddTool(getAppByNameTool, getAppByNameToolmpl)
+
+	getAppByClientIdTool, getAppByClientIdToolmpl := tools.GetApplicationByClientIdTool()
+	s.AddTool(getAppByClientIdTool, getAppByClientIdToolmpl)
```

Comment:
When more tools are added, main.go file length will be increased which wouldn't be scalable. @shashimalcse any suggestions to address this?

Reply:
IMO, we have to structure this to different tool collections (app tools, user tools, api resource tools) and pass in the server.
Let's structure as we hit the problem